pkg/ccl/changefeedccl/metrics.go

```diff
@@ -174,6 +190,57 @@ func (m *sliMetrics) getBackfillRangeCallback() func(int64) (func(), func()) {
 	}
 }
 
+type wrappingCostController struct {
+	ctx      context.Context
+	inner    metricsRecorder
```

Comment:
if you embed inner recorder, you won't have to override noop methods below -- just the messages callback.

Reply:
I didn't do this on purpose so that we have to decide explicitly when new methods are added whether or not they need accounting.